cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if t.taskID != "" {
+		id, err := influxdb.IDFromString(t.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return t.printTasks(taskPrintOpts{tasks: tasks})
 }
 
-var taskUpdateFlags struct {
-	id     string
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (t *cmdTaskBuilder) taskRerunFailedCmd() *cobra.Command {
+	cmd := t.opts.newCmd("rerun_failed", t.taskRerunFailedF, true)
+	cmd.Short = "Find and Rerun failed runs/tasks"
+	cmd.Aliases = []string{"rrf"}
+
+	t.org.register(t.opts.viper, cmd, false)
+	t.globalFlags.registerFlags(t.opts.viper, cmd)
+	registerPrintOptions(t.opts.viper, cmd, &t.taskPrintFlags.hideHeaders, &t.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&t.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&t.taskRerunFailedFlags.before, "before", "", "before interval")
+	cmd.Flags().StringVar(&t.taskRerunFailedFlags.after, "after", "", "after interval")
+
+	return cmd
+}
+
+func (t *cmdTaskBuilder) taskRerunFailedF(*cobra.Command, []string) error {
+	if err := t.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+
+	tskSvc, _, err := t.svcFn()
+	if err != nil {
+		return err
+	}
+
+	taskIDPresent := t.taskID == ""
+
+	/*
+		If no TaskID is given, use TaskFilter to get all Tasks, search for failed runs in each, then re run failed
+		If TaskID given, use RunFilter to search for failed runs then re run
+	*/
+	taskFilter := influxdb.TaskFilter{}
+	runFilter := influxdb.RunFilter{}
+	if !taskIDPresent {
+		if t.org.name != "" {
+			taskFilter.Organization = t.org.name
+		}
+		if t.org.id != "" {
+			orgID, err := influxdb.IDFromString(t.org.id)
+			if err != nil {
+				return err
+			}
+			taskFilter.OrganizationID = orgID
+		}
+	} else {
+		id, err := influxdb.IDFromString(t.taskID)
+		if err != nil {
+			return err
+		}
+		runFilter.Task = *id
+	}
+
+	runFilter.BeforeTime = t.taskRerunFailedFlags.before
+	runFilter.AfterTime = t.taskRerunFailedFlags.after
+
+	var allRuns []*influxdb.Run
+	if !taskIDPresent {
+		allTasks, _, err := tskSvc.FindTasks(context.Background(), taskFilter)
+		if err != nil {
+			return err
+		}
+
+		for _, t := range allTasks {
+			runFilter.Task = t.ID
+			runsPerTask, _, err := tskSvc.FindRuns(context.Background(), runFilter)
+			if err != nil {
+				return err
+			}
+			allRuns = append(allRuns, runsPerTask...)
```

Comment:
I would filter to only the failed runs at this layer, in case the total size of all runs is very big.